src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectDefaultConverter.cs

```diff
@@ -143,6 +144,8 @@ internal override bool OnTryRead(ref Utf8JsonReader reader, Type typeToConvert,
 
                     state.Current.ReturnValue = obj;
                     state.Current.ObjectState = StackFrameObjectState.CreatedObject;
+
+                    jsonTypeInfo.InitializeRequiredProperties(ref state);
```

Comment:
This seems like a method that would be better suited for `WriteStackFrame`, i.e. something like
```C#
state.Current.InitializeRequiredPropertyState(jsonTypeInfo);
```

Reply:
The reason I added here is to have all implementation pieces together rather than spread around multiple files. I think it makes code easier to read this way

Reply:
The `MarkRequiredPropertyAsRead` method is already defined on `ReadStackFrame`, so arguably they're spread out right now? I'd recommending moving all 3 methods to `ReadStackFrame`, since this is where we generally host methods that modify the serialization state.

Reply:
that works to